src/pybind/mgr/dashboard/frontend/src/app/shared/api/prometheus.service.ts

```diff
@@ -163,11 +163,9 @@ export class PrometheusService {
                 checkNan
               ) {
                 queriesResults[queryName].forEach((valueArray: any[]) => {
-                  valueArray.forEach((val, index) => {
-                    if (isNaN(parseFloat(val[1]))) {
-                      valueArray[index][1] = '0';
-                    }
-                  });
+                  if (isNaN(parseFloat(valueArray[1])) || valueArray[1] === 'NaN') {
+                    valueArray[1] = '0';
+                  }
```

Comment:
This change is required because in the existing implementation we are parsing the `valueArray` in a double loop but on investigation I found out that a single loop is required to parse the array values, so the existing implementation stands incorrect. Verified with plotting some data in the graphs and works as expected with the incoming change.